dart/lib/src/client.dart

```diff
@@ -64,88 +37,19 @@ abstract class SentryClient {
   /// * https://docs.sentry.io/learn/context/#capturing-the-user
   User userContext;
 
-  /// Use for browser stacktrace
-  String origin;
-
-  /// Used by sentry to differentiate browser from io environment
-  final String _platform;
-
-  final Sdk sdk;
-
-  String get clientId => sdk.identifier;
-
-  @visibleForTesting
-  String get postUri {
-    final port = dsnUri.hasPort &&
-            ((dsnUri.scheme == 'http' && dsnUri.port != 80) ||
-                (dsnUri.scheme == 'https' && dsnUri.port != 443))
-        ? ':${dsnUri.port}'
-        : '';
-    final pathLength = dsnUri.pathSegments.length;
-    String apiPath;
-    if (pathLength > 1) {
-      // some paths would present before the projectID in the dsnUri
-      apiPath =
-          (dsnUri.pathSegments.sublist(0, pathLength - 1) + ['api']).join('/');
-    } else {
-      apiPath = 'api';
-    }
-    return '${dsnUri.scheme}://${dsnUri.host}$port/$apiPath/$projectId/store/';
-  }
-
   /// Reports an [event] to Sentry.io.
   Future<SentryId> captureEvent(
     SentryEvent event, {
     Scope scope,
     dynamic hint,
   }) async {
-    final now = options.clock();
-    var authHeader = 'Sentry sentry_version=6, sentry_client=$clientId, '
-        'sentry_timestamp=${now.millisecondsSinceEpoch}, sentry_key=$publicKey';
-    if (secretKey != null) {
-      authHeader += ', sentry_secret=$secretKey';
-    }
-
-    final headers = buildHeaders(authHeader);
-
-    final data = <String, dynamic>{
-      'project': projectId,
-      'event_id': event.eventId.toString(),
-      'timestamp': formatDateAsIso8601WithSecondPrecision(event.timestamp),
-    };
-
-    if (options.environmentAttributes != null) {
-      mergeAttributes(options.environmentAttributes.toJson(), into: data);
-    }
+    event = _processEvent(event, eventProcessors: options.eventProcessors);
 
-    // Merge the user context.
-    if (userContext != null) {
-      mergeAttributes(<String, dynamic>{'user': userContext.toJson()},
-          into: data);
-    }
-
-    mergeAttributes(
-      event.toJson(
-        origin: origin,
-      ),
-      into: data,
-    );
-    mergeAttributes(<String, String>{'platform': _platform}, into: data);
-
-    final body = bodyEncoder(data, headers);
-
-    final response = await options.httpClient.post(
-      postUri,
-      headers: headers,
-      body: body,
-    );
+    event = _applyScope(event: event, scope: scope);
 
-    if (response.statusCode != 200) {
-      return SentryId.empty();
-    }
+    event = event.copyWith(platform: transport.platform);
```

Comment:
why do we need to read the `platform` from the `transport`? its already defining on the SentryEvent ctor based on `isWeb`.
I believe `transport.platform` could be done altogether

Reply:
✅